src/site/stages/build/plugins/create-react-pages.js

```diff
@@ -1,54 +1,35 @@
-
-function createTemporaryReactPages(options) {
+function createReactPages() {
   return (files, metalsmith, done) => {
-    if (!prodEnvironments.has(options.buildtype)) {
-      const root = path.join(__dirname, '../../../../..');
-      manifestHelpers
-        .getAppManifests(root)
-        // Skip manifests that are mapping the url from settings.js
-        .filter(m => m.rootUrl)
-        .forEach(({ entryName, appName, rootUrl }) => {
-          // Removing leading and trailing slashes
-          const trimmedUrl = rootUrl.endsWith('/')
-            ? rootUrl.substring(1, rootUrl.length - 1)
-            : rootUrl.substring(1);
-          const filePath = `${trimmedUrl}/index.html`;
+    const root = path.join(__dirname, '../../../../..');
+    manifestHelpers
+      .getAppManifests(root)
+      // Skip manifests that are mapping the url from settings.js
+      .filter(m => m.rootUrl)
+      .forEach(({ entryName, appName, rootUrl, template }) => {
+        const trimmedUrl = path.join('.', rootUrl);
+        const filePath = path.join(trimmedUrl, 'index.html');
 
-          if (!files[filePath]) {
-            console.log(`Creating temp React landing page: ${rootUrl}`);
-            files[filePath] = {
-              title: appName,
-              entryname: entryName,
-              layout: 'page-react.html',
-              path: trimmedUrl,
-              contents: new Buffer(`
-              <div class="usa-alert usa-alert-warning usa-alert-full-width background-color-only vads-u-margin-top--0 vads-u-padding-y--1">
-                <div class="usa-alert-body">
-                  <div class="vads-l-grid-container">
-                    <strong>Non-production test page:</strong> This is a temporary page generated by Metalsmith for testing on dev and staging. You will need to create a page in Drupal or vagov-content repo to make it available in production.
-                  </div>
-                </div>
-              </div>
-              <nav aria-label="Breadcrumb" aria-live="polite" class="va-nav-breadcrumbs" id="va-breadcrumbs">
-                <ul class="row va-nav-breadcrumbs-list columns" id="va-breadcrumbs-list">
-                  <li><a href="/">Home</a></li>
-                  <li><a aria-current="page" href="${rootUrl}">${appName}</a></li>
-                </ul>
-              </nav>
-              `),
-            };
-          }
-        });
-    }
+        if (!files[filePath]) {
+          console.log(
+            `Generating HTML template for application ${appName} at ${rootUrl}`,
+          );
+          files[filePath] = {
+            title: appName,
+            entryname: entryName,
+            path: trimmedUrl,
+            layout: 'page-react.html',
+            contents: Buffer.from('\n<!-- Generated from manifest.json -->\n'),
+            ...template,
+          };
+        }
+      });
 
     done();
   };
 }
 
-module.exports = createTemporaryReactPages;
+module.exports = createReactPages;
```

Comment:
@micahchiang This is the change for enabling manifest.json's to do what we previously required vagov-content for. It's not a lot, because our build already generated "temporary" React landing pages a similar way, but adds support for the `template` property.

Reply:
Cool, thanks for tagging me on this!